plugin-build/src/main/kotlin/io/sentry/android/gradle/autoinstall/okhttp/OkHttpInstallStrategy.kt

```diff
@@ -0,0 +1,69 @@
+package io.sentry.android.gradle.autoinstall.okhttp
+
+import io.sentry.android.gradle.SentryPlugin
+import io.sentry.android.gradle.autoinstall.AutoInstallState
+import io.sentry.android.gradle.autoinstall.InstallStrategyRegistrar
+import io.sentry.android.gradle.autoinstall.SENTRY_GROUP
+import io.sentry.android.gradle.util.SemVer
+import io.sentry.android.gradle.util.info
+import io.sentry.android.gradle.util.warn
+import javax.inject.Inject
+import org.gradle.api.artifacts.ComponentMetadataContext
+import org.gradle.api.artifacts.ComponentMetadataRule
+import org.gradle.api.artifacts.dsl.ComponentMetadataHandler
+import org.slf4j.Logger
+
+// @Inject is needed to avoid Gradle error
+// @CacheableRule
+abstract class OkHttpInstallStrategy @Inject constructor() : ComponentMetadataRule {
+
+    private var logger: Logger = SentryPlugin.logger
+
+    constructor(logger: Logger) : this() {
+        this.logger = logger
+    }
+
+    override fun execute(context: ComponentMetadataContext) {
```

Comment:
Can we somehow DRY for the `execute` method? OkHttp and others look like the same, maybe we get the configuration in a generic way?

Reply:
Yeah I think we can, will try to come up with something